doctr/models/predictor/pytorch.py

```diff
@@ -57,6 +64,13 @@ def forward(
         if any(page.ndim != 3 for page in pages):
             raise ValueError("incorrect input shape: all pages are expected to be multi-channel 2D images.")
 
+        origin_page_shapes = [page.shape[:2] if isinstance(page, np.ndarray) else page.shape[-2:] for page in pages]
+
+        # Detect document rotation and rotate pages
+        if self.straighten_pages:
+            origin_page_orientations = [estimate_orientation(page) for page in pages]
+            pages = [rotate_image(page, -angle, expand=True) for page, angle in zip(pages, origin_page_orientations)]
```

Comment:
I think you'll have to pass `preserve_origin_shape=True` over here to fix the failing test :)

Reply:
I don't get it :( this test passes on my computer ... I will have a look !

Reply:
seems to be good now